File: server/dto/action_callback_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.coordinates_dto import CoordinatesDto
from dto.game_context_dto import GameContextDto
from dto.match_action_dto import MatchActionDto
from dto.partial_spell_dto import PartialSpellDto
from game_engine.models.actions.callbacks.action_callback_id import ActionCallBackId


@dataclass
class ActionCallbackDto(BaseDto):
    id: int
    parentAction: MatchActionDto
    parentCallbackId: ActionCallBackId
    spellCause: PartialSpellDto | None
    impactedCoords: list[CoordinatesDto] | None
    updatedGameContext: GameContextDto

File: client/src/dto/ActionCallbackDto.ts
import { ActionCallbackId } from "../enums/actionCallbackId";
import { CoordinatesDto } from "./CoordinatesDto";
import { GameContextDto } from "./GameContextDto";
import { MatchActionDto } from "./MatchActionDto";
import { PartialSpellDto } from "./PartialSpellDto";

export interface ActionCallbackDto {
    id: ActionCallbackId;
    parentAction: MatchActionDto;
    parentCallbackId: ActionCallbackId;
    spellCause: PartialSpellDto | null;
    impactedCoords: CoordinatesDto | null;
    updatedGameContext: GameContextDto;
}